File: backend/catalog-service/core_test/type/domain/type_mother.py
from typing import List

from core.shared.base.domain.uuid import UUID
from core.type.domain.type import Type
from core.type.domain.type_id import TypeId


class TypeMother:
    @staticmethod
    def random_id() -> TypeId:
        return TypeId(
            UUID.random()
        )

    @staticmethod
    def random_types() -> List[Type]:
        type_list = [
            Type(
                type_id=TypeMother.random_id(),
                name='any_type_1'
            ),
            Type(
                type_id=TypeMother.random_id(),
                name='any_type_2'
            )
        ]

        return type_list

File: frontend/web-app/src/core_test/type/domain/TypeMother.ts
import {Type} from "../../../core/type/domain/Type";
import {UUID} from "../../../core/shared/domain/UUID";

export class TypeMother {

    static randomType(): Type {
        return {
            typeId: UUID.random(),
            name: 'Any Name'
        }
    }

    static randomTypes(number: number): Type[] {
        const types: Type[] = [];
        for (let i = 0; i < number; i++) {
            types.push(TypeMother.randomType());
        }

        return types;
    }
}
